lab_4_summarization_textrank/main.py

```diff
@@ -120,23 +170,42 @@ def __init__(self) -> None:
         """
         Constructs all the necessary attributes
         """
-        pass
+        super().__init__()
+        self._word2id = {}
+        self._id2word = {}
+        self._last_ident = 0
 
     def _learn_indices(self, tokens: tuple[str, ...]) -> None:
         """
         Fills attributes mapping words and integer equivalents to each other
         :param tokens: a sequence of string tokens
         :return:
         """
-        pass
+        check_type(tokens, tuple, str)
+
+        for idx, token in enumerate(tokens, start=1000):
```

Comment:
with `start=1000` you assign the same identifiers multiple times (depends on how many times the function was called)

Reply:
i don't really get what you mean :(

Reply:
some tokens are repeated but in the end every one of them appears in the dictionary only once, as in case a key is already in the dictionary, its value is just overwritten??
`for idx, token in enumerate(tokens, start=1000):
       self._word2id[token] = idx`

Reply:
I like the new version better but I'm still unsure

let's see an example 

1) first call: passed 100 unique words -> assigned ids from 1000 to 1100 
2) second call: passed 200 unique words, but 30 of them are already present in the dict -> assigned ids from 1100 to 1300, and there are 100 + 200 - 30 = 270 words in the dict now
3_ third call: passed 50 more words -> assigned ids from 1270 to 1320, but !!!! on the previous iteration we also assigned such ids as 1270, 1271 etc... 

do you maybe want to use `_last_ident`?

Reply:
yeah, got it! i was wondering what i should do with that variable, now i understand